test/e2e/cluster_upgrade_runtimesdk.go

```diff
@@ -694,30 +754,89 @@ func beforeClusterUpgradeTestHandler(ctx context.Context, c client.Client, clust
 			}
 		}
 
+		mps := framework.GetMachinePoolsByCluster(ctx,
+			framework.GetMachinePoolsByClusterInput{ClusterName: cluster.Name, Namespace: cluster.Namespace, Lister: c})
+		for _, mp := range mps {
+			if mp.Spec.Template.Spec.Version != fromVersion {
+				return false
+			}
+		}
+
+		// BeforeClusterUpgrade is followed by BeforeControlPlaneUpgrade.
```

Comment:
Can we make it a bit clearer that this here is checking that BeforeControlPlaneUpgrade is not yet called when we block?